jiminy/src/lib.rs

```diff
@@ -0,0 +1,31 @@
+use core::fmt::Display;
+use inf1_core::quote::swap::err::SwapQuoteErr;
+use inf1_ctl_jiminy::{err::Inf1CtlErr, program_err::Inf1CtlCustomProgErr};
+use jiminy_log::sol_log;
+use jiminy_program_error::ProgramError;
+
+pub struct SwapQuoteProgErr<I, O, P>(pub SwapQuoteErr<I, O, P>);
+
+impl<
+        I: core::fmt::Debug + Display,
+        O: core::fmt::Debug + Display,
+        P: core::fmt::Debug + Display,
+    > From<SwapQuoteProgErr<I, O, P>> for ProgramError
+where
+    ProgramError: From<I> + From<O> + From<P>,
```

Comment:
- doesnt seem like the `Debug` trait bound is used here (`Display` might be used by the `.to_string()`)
- in general you want to bound traits using `Into` where possible because `Into` is more permissible than `From`

More info: https://doc.rust-lang.org/rust-by-example/conversion/from_into.html#from-and-into-are-interchangeable

```suggestion
impl<
        I: Display + Into<ProgramError>,
        O: Display + Into<ProgramError>,
        P: Display + Into<ProgramError>,
    > From<SwapQuoteProgErr<I, O, P>> for ProgramError
```